src/connector/Cargo.toml

```diff
@@ -37,7 +37,7 @@ nexmark = { path = "../utils/nexmark", features = ["serde"] }
 num-traits = "0.2"
 paste = "1"
 prost = "0.11"
-pulsar = { version = "5", default-features = false, features = ["tokio-runtime"], rev = "7fab6a9", git = "https://github.com/skyzh/pulsar-rs" }
+pulsar = "4.1.3"
```

Comment:
I guess you need to disable the default features.

Reply:
seem true. The release notes claim that they are compatible with tokio and async-std